Sources/ApolloCodegenLib/Frontend/JavaScript/src/compiler/index.ts

```diff
@@ -71,6 +72,17 @@ export function compileToIR(
   const fragmentMap = new Map<String, ir.FragmentDefinition>();
   const referencedTypes = new Set<GraphQLNamedType>();
 
+  const queryType = schema.getQueryType() as GraphQLNamedType;
+  if (queryType === undefined) {
+    throw new GraphQLError("GraphQL Schema must contain a 'query' root type definition.", { });
```

Comment:
I don't think this we'd ever reach this stage. Since this is a spec violation it'll likely get caught by graphql-js validation before we start compiling the IR.